Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -278,12 +307,15 @@ function reduce(reducer /*, initialValue */)
     if (!@isObject(this))
         @throwTypeError("Iterator.prototype.reduce requires that |this| be an Object.");
 
-    if (!@isCallable(reducer))
+    var iterated = this;
```

Comment:
Let's just use `this` directly in this function instead of `iterated` when possible.